packages/loader/container-loader/src/container.ts

```diff
@@ -1312,6 +1320,12 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     private async rehydrateDetachedFromSnapshot(detachedContainerSnapshot: ISummaryTree) {
+        const detachedBlobSummary = detachedContainerSnapshot.tree[".detachedBlobStorage"];
+        if (detachedBlobSummary !== undefined) {
+            assert(!!this.loader.services.detachedBlobStorage, "detached blob summary but no detached blob storage");
+            this.loader.services.detachedBlobStorage.rehydrate(detachedBlobSummary as ISummaryTree);
+        }
```

Comment:
I'm not sure I follow the purpose of this logic. Assuming that we want to support 400Mb payloads, this flow will move memory (duplicating it on the way) and make FF intermediate between two pieces of code in host. What is the goal?
Also why it is ISummaryTree? Feels like unknown is much better type, as FF code does not do anything with it, just passes around. ISummaryTree is internal format, host should not have any knowledge or dependency on it

Reply:
So we should expect host to serialize/rehydrate IDetachedBlobStorage without Container being involved?

Reply:
I'm not sure what you mean by "serialize" here. I'd think we want to have generic system that supports calling Container.serialize() multiple times in a row, possibly in rapid flow. And I'd expect the process to be as efficient as possible, i.e. if 100Mb blob was created before first serialize() call, and is still reused on the last call, then its impact to the cost of subsequent serialize() call to be zero (as nothing really changed RE usage of such blob).

Given that we do not provide any kind of GC or notification to the implementer of IDetachedBlobStorage what blobs are still in use in latest snapshot, I think the logical conclusion would be that result of IDetachedBlobStorage.createBlob() have to be valid until host either destroys container instance and all its side effects (detached summaries), or calls Container.attach().
How host adheres to this requirement of keeping blobs around - it's really up to host (i.e. store it in memory, or in longer term storage like indexDB), but that's host problem to solve. Most likely it would do same thing for result of Container.serialize() and IDetachedBlobStorage.createBlob() - i.e. either both results are only stored in memory (and everything is lost on app reboot), or it would store both in some longer term storage.

At least that's my mental model